File: output/ChatGPT/avatar/Java/C++/codeforces_306_A.cpp
#include <iostream>
using namespace std;

int main() {
    int n, m;
    cin >> n >> m;
    int a[m];

    if (n % m == 0) {
        for (int i = 0; i < m; i++) {
            a[i] = n / m;
        }
    } else {
        int sub = n / m;
        for (int i = 0; i < m; i++) {
            a[i] = sub;
        }
        int test = n - (sub * m);
        int count = 0;
        for (int i = 0; i < test; i++) {
            a[count] = a[count] + 1;
            count++;
            if (count >= m) count = 0;
        }
    }

    for (int i = m - 1; i >= 0; i--) {
        cout << a[i] << " ";
    }

}

File: output/ChatGPT/avatar/Java/Python/codeforces_306_A.py
n, m = map(int, input().split())
a = []

if n % m == 0:
    a = [n // m] * m
else:
    sub = n // m
    a = [sub] * m
    test = n - (sub * m)
    count = 0
    for i in range(test):
        a[count] += 1
        count += 1
        if count >= len(a):
            count = 0

print(*a[::-1])  # Print the elements of a in reverse order